src/test/java/com/github/hanyaeger/api/engine/entities/entity/YaegerEntityTest.java

```diff
@@ -624,170 +631,18 @@ void angleToNullYaegerEntityThrowsNullPointerException() {
-        // Arrange
-        var expected = Direction.RIGHT.getValue();
-        var other = new YaegerEntityImpl(new Coordinate2D(LOCATION.getX() + 10, LOCATION.getY()));
-        var otherNode = mock(Node.class, withSettings().withoutAnnotations());
-        other.setNode(Optional.of(otherNode));
-        var otherBoundingBox = mock(BoundingBox.class);
-        when(otherNode.getBoundsInLocal()).thenReturn(otherBoundingBox);
-        when(otherBoundingBox.getWidth()).thenReturn(ENTITY_WIDTH);
-        when(otherBoundingBox.getHeight()).thenReturn(ENTITY_HEIGHT);
-        when(otherNode.getScene()).thenReturn(scene);
-        when(scene.getWidth()).thenReturn(SCENE_WIDTH);
-
-        other.init(injector);
-        other.transferCoordinatesToNode();
-
-        sut.init(injector);
-        sut.transferCoordinatesToNode();
-
-        // Act
-        var actual = sut.angleTo(other);
-
-        // Assert
-        assertEquals(expected, actual);
-    }
-
-    @Test
-    void angleToYaegerEntityLeftAboveIs225Degrees() {
-        // Arrange
-        var expected = 225d;
-        var other = new YaegerEntityImpl(new Coordinate2D(LOCATION.getX() - 10, LOCATION.getY() - 10));
-        var otherNode = mock(Node.class, withSettings().withoutAnnotations());
-        other.setNode(Optional.of(otherNode));
-        var otherBoundingBox = mock(BoundingBox.class);
-        when(otherNode.getBoundsInLocal()).thenReturn(otherBoundingBox);
-        when(otherBoundingBox.getWidth()).thenReturn(ENTITY_WIDTH);
-        when(otherBoundingBox.getHeight()).thenReturn(ENTITY_HEIGHT);
-        when(otherNode.getScene()).thenReturn(scene);
-        when(scene.getWidth()).thenReturn(SCENE_WIDTH);
-
-        other.init(injector);
-        other.transferCoordinatesToNode();
-
-        sut.init(injector);
-        sut.transferCoordinatesToNode();
-
-        // Act
-        var actual = sut.angleTo(other);
-
-        // Assert
-        assertEquals(expected, actual);
-    }
-
-    @Test
-    void angleToYaegerEntityRightAboveIs135Degrees() {
-        // Arrange
-        var expected = 135d;
-        var other = new YaegerEntityImpl(new Coordinate2D(LOCATION.getX() + 10, LOCATION.getY() - 10));
-
-        other.init(injector);
-        other.transferCoordinatesToNode();
-        other.applyTranslationsForAnchorPoint();
-
-        sut.transferCoordinatesToNode();
-        sut.applyTranslationsForAnchorPoint();
-
-        // Act
-        var actual = sut.angleTo(other);
-
-        // Assert
-        assertEquals(expected, actual);
-    }
-
-    @Test
-    void angleToYaegerEntityLeftBelowIs315Degrees() {
+    @ParameterizedTest
+    @MethodSource("provideArgumentsForAngleTo")
+    void testAngleToOtherEntity(Coordinate2D otherLocation, double expectedAngle) {
         // Arrange
-        var expected = 315d;
-        var other = new YaegerEntityImpl(new Coordinate2D(LOCATION.getX() - 10, LOCATION.getY() + 10));
+        var other = new YaegerEntityImpl(otherLocation);
```

Comment:
We should have some tests that explicitly test the case in which we use a composite entity